### PR TITLE
Simplify docker compose and dsh

### PR DESCRIPTION
**Pull request by acbramley**

### CODE REVIEW

**Comment by pingers** [comment]
It's having a whinge about /shared/tmp not existing and setting permissions on /shared/private and public before they exist.. I'll continue in the morning.

**Comment by pingers** [comment]
Oh, another thing to consider is this breaks CAS login due to whitelisting wildcard domains for development... but not 127.0.0.1. *Sigh*

**Comment by pingers** [comment]
Also need to test Shepherd with this.. maaybe that will just run web container on a different port. I.e. OpenShift router will use port 80. We'll see.

File: dsh [hunk 1]
```diff
@@ -4,17 +4,6 @@
...
-# Set the top level domain used to by nginx-proxy to direct traffic to
-# conatiners. The TLDs 'local' and 'localhost' are reserved in macOS, and 'dev'
-# exists in global DNS, which may cause issues reassigning it with dnsmasq.
-export DOMAIN=${DOMAIN:-'test'}
-
-# Nginx proxy exposed port. Useful when using something other than 80.
-export NGINX_PORT=$(docker port nginx-proxy | awk -F ':' '{print $2}')
-
 # Used as the prefix for docker networking, container naming and nginx hostname.
 export PROJECT=$(basename ${PWD})
```
[the remark this answers is not in the document]
Comment (acbramley) [reply dsh]: It's used in shell and status to check that the containers are running?

File: dsh [hunk 2]
```diff
@@ -163,230 +100,11 @@ dsh_project() {
...
-  fi
-}
 
 # Command: ./dsh pull
 # Fetches all images used by the project.
 dsh_pull() {
-  docker-compose pull
+  docker-compose -f ${DOCKER_COMPOSE_FILE} pull
   docker pull jwilder/nginx-proxy
   docker pull uofa/utility-php7
```
[the remark this answers is not in the document]
Comment (acbramley) [reply dsh]: Nice!

File: dsh [hunk 3]
```diff
@@ -78,59 +62,20 @@ dsh_shell() {
...
-    docker exec ${PROJECT}_web_1 /bin/bash -c "echo '${USER} ALL=(ALL) NOPASSWD: ALL' > /etc/sudoers.d/${USER}"
-  fi
-
-  docker exec --interactive --tty --user ${USER} \
-    --env SSH_AUTH_SOCK=${CONTAINER_SSH_AUTH_SOCK} \
-    --env LINES=$(tput lines) --env COLUMNS=$(tput cols) \
-    ${DSH_PROJECT[@]:-} \
-    --env TERM=xterm-256color \
-    ${PROJECT}_web_1 ${@:-/bin/bash}
+  docker-compose -f ${DOCKER_COMPOSE_FILE} exec web /bin/bash
```
[the remark this answers is not in the document]
Comment (acbramley) [reply dsh]: Hmm adding it back doesn't seem to be working right, fails with error:

`OCI runtime exec failed: exec failed: container_linux.go:348: starting container process caused "exec: \"drush pml\": executable file not found in $PATH": unknown`

File: dsh (hunk 1 above)
[the remark this answers is not in the document]
Comment (pingers) [reply dsh]: Ooookay, it can stay :+1:

File: dsh (hunk 3 above)
[the remark this answers is not in the document]
Comment (singularo) [reply dsh]: Ahh, this issue is why I've put the path into the env in https://github.com/universityofadelaide/shepherd-drupal-scaffold/pull/39 - If you don't run bash, it doesn't do the path.

[the remark this answers is not in the document]
Comment (acbramley) [reply dsh]: It's something to do with how we're passing along the variables.
`docker-compose -f docker-compose.osx.yml exec web ./bin/drush pml` works
but
`./dsh shell ./bin/drush pml` doesn't

**Comment by fenetikm** [comment]
Getting the following on a fresh install:
```
Project not running, starting.
Starting project containers.
Creating network "ua-wcms-foundation_default" with the default driver
Creating ua-wcms-foundation_xdebug_1 ... done
Creating ua-wcms-foundation_web_1    ... done
Creating ua-wcms-foundation_mail_1   ...
Creating ua-wcms-foundation_db_1     ...
Creating ua-wcms-foundation_mail_1   ... error

ERROR: for ua-wcms-foundation_redis_1  Cannot start service redis: cannot join network of a non running container: 4cb69cea736d4ae339ef880de8ffff8cb6321802116542bfc331958f0b1c0f14
Creating ua-wcms-foundation_db_1     ... error
ERROR: for ua-wcms-foundation_mail_1  Cannot start service mail: cannot join network of a non running container: 4cb69cea736d4ae339ef880de8ffff8cb6321802116542bfc331958f0b1c0f14

ERROR: for ua-wcms-foundation_db_1  Cannot start service db: cannot join network of a non running container: 4cb69cea736d4ae339ef880de8ffff8cb6321802116542bfc331958f0b1c0f14

ERROR: for redis  Cannot start service redis: cannot join network of a non running container: 4cb69cea736d4ae339ef880de8ffff8cb6321802116542bfc331958f0b1c0f14

ERROR: for mail  Cannot start service mail: cannot join network of a non running container: 4cb69cea736d4ae339ef880de8ffff8cb6321802116542bfc331958f0b1c0f14

ERROR: for db  Cannot start service db: cannot join network of a non running container: 4cb69cea736d4ae339ef880de8ffff8cb6321802116542bfc331958f0b1c0f14
ERROR: Encountered errors while bringing up the project.
==>! docker-compose --version                                                                                                                                                                                                  develop ✗ ~ ?  ~/$/UofA/shepherd/github/ua-wcms-foundation
docker-compose version 1.22.0, build f46880f
```

Any thoughts? Cheers.

**Comment by acbramley** [comment]
@fenetikm I haven't seen that before, I also got another PNX dev up and running on D4M yesterday and we had no problems. Can you try:

`docker-composer -f docker-compose.osx.yml down`
`docker network prune`
`docker-compose -f docker-compose.osx.yml up`

**Comment by fenetikm** [comment]
Thanks for the idea @acbramley but no luck, same problem. Might be the way I have docker for mac installed as I was stuffing around with it regarding open shift etc. I think I will reinstall that and see if that sorts it. Cheers.

**Comment by acbramley** [comment]
If you get the same can you paste in the output of

`docker-compose -f docker-compose.osx.yml ps` and ``docker-compose -f docker-compose.osx.yml logs web`

**Comment by fenetikm** [comment]
```
==>! docker-compose -f docker-compose.osx.yml ps                                                                                                                                                                               develop ✗ ~ ?  ~/$/UofA/shepherd/github/ua-wcms-foundation
           Name                          Command                State     Ports
-------------------------------------------------------------------------------
ua-wcms-foundation_db_1       docker-entrypoint.sh mysqld      Exit 128
ua-wcms-foundation_mail_1     MailHog                          Exit 128
ua-wcms-foundation_redis_1    docker-entrypoint.sh redis ...   Exit 128
ua-wcms-foundation_web_1      /apache2-foreground              Exit 1
ua-wcms-foundation_xdebug_1   /usr/local/bin/d4m-tcp-for ...   Up
```

**Comment by fenetikm** [comment]
nice one - sorted.

**Comment by pingers** [comment]
This is ready to ship. Works for everyone at this point :+1:

File: dsh (hunk 3 above)
[the remark this answers is not in the document]
Comment (acbramley) [reply dsh]: With some help from some PNX guys, debugging the bash script shows that it's quoting the string, e.g it's running

`docker-compose -f docker-compose.osx.yml exec -e COLUMNS=206 -e LINES=43 web './bin/drush pml'`

[the remark this answers is not in the document]
Comment (pingers) [reply dsh]: Cool!

**Pull request merged**